haskell/DemystifyingFoldable.hs

```diff
@@ -0,0 +1,45 @@
+{-
+
+Foldable represents a structure that, given a suitable "folding" function, can
+be collapsed into a new value without the structure.
+
+In these reps you'll practice the various ways Foldable can be implemented and
+the how to implement it on different kinds of datastructures.
+
+- Implement a type like Maybe
+- Write a full instance of Foldable for your "Maybe"
+- Write a newtype around List
+- Write a full instance of Foldable for your "List"
```

Comment:
What do you mean by "full instance" here? There are lots more methods of `Foldable` that you _could_ implement, but provide defaults. Based on the instances you provide below, I would probably just say "with explicitly implementations for `foldr` and `foldMap`" or something like that.

Reply:
Ah, thanks for pointing that out. I made the assumption that this was the full instance since either of those two makes for a minimal instance. That was a silly assumption. I'll update the language.